stdlib/io.pyi

```diff
@@ -67,11 +68,7 @@ class FileIO(RawIOBase, BinaryIO):
     # Technically this is whatever is passed in as file, either a str, a bytes, or an int.
     name: Union[int, str]  # type: ignore
     def __init__(
-        self,
-        file: Union[str, bytes, int],
-        mode: str = ...,
-        closefd: bool = ...,
-        opener: Optional[Callable[[Union[int, str], str], int]] = ...,
+        self, file: Union[str, bytes, int], mode: str = ..., closefd: bool = ..., opener: Optional[_Opener] = ...
```

Comment:
actually this is generic:

```
T_filelike = TypeVar("T_filelike", str, bytes, os.PathLike)

@overload
def __init__(self, file: int, opener: None = ...) -> None: ...
@overload
def __init__(self, file: T_filelike, opener=Optional[Callable[[T_filelike, int], int]) -> None: ...
```

Reply:
I've been using:

```python
io.FileIO(object(), opener=lambda *args: breakpoint())
```

and

```python
_pyio.FileIO(object(), opener=lambda *args: breakpoint())
```

Reply:
The `name` attribute might need some updating too:

```
>>> io.FileIO(pathlib.Path("/tmp/foo"), "w").name
PosixPath('/tmp/foo')
```

I think it would be more practical to use unions instead of a generic. Code using the `name` attribute already needs to handle integers, so it makes sense to do `str(fileio.name)` and adding more path-like members to the union doesn't really change this in any way. On the other hand, if we make it generic, then any code using `FileIO` has to be changed to do `FileIO[str]` or whatever.

There are some handy `PathLike` aliases in `_typeshed` that we can use.

Reply:
@Akuli can I ask you to take over this one please?